### PR TITLE
feature/AT-UI-BE 009 Created GET Operation

### PR DESCRIPTION
**Pull request by ErnestoZamora**
Created GET Operation

### CODE REVIEW

File: src/main/java/com/university/university/controller/UniversityController.java [hunk 1]
```diff
@@ -1,23 +1,38 @@
...
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.data.mongodb.repository.MongoRepository;
 
 import lombok.extern.slf4j.Slf4j;
 
-@RequestMapping(value = "/university/")
+import java.util.List;
+import java.util.Optional;
```
Comment (JoseCortes-Agile) [inline src/main/java/com/university/university/controller/UniversityController.java]: Please remove any Import that is not used

File: src/main/java/com/university/university/controller/UniversityController.java [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
+	@Autowired
+	private AtUniversityServiceImpl service;
+
+	@GetMapping(value = "/course")
 	@ResponseStatus(HttpStatus.OK)
-	public String getOperation() {
+	public List getOperation() {
 		log.info("Calling Get Operation");
-		return "Hello at-university-api";
+		//return "Hello at-university-api";
```
Comment (JoseCortes-Agile) [inline src/main/java/com/university/university/controller/UniversityController.java]: Please remove commented code, we should deliver clean code

File: src/main/java/com/university/university/model/Course.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.NoArgsConstructor;
+import org.bson.types.ObjectId;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.annotation.PersistenceConstructor;
+import org.springframework.data.mongodb.core.mapping.Document;
+import org.springframework.data.mongodb.repository.Query;
+import org.springframework.data.repository.RepositoryDefinition;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Indexed;
```
Comment (JoseCortes-Agile) [inline src/main/java/com/university/university/model/Course.java]: Remove all the unused imports

File: src/main/java/com/university/university/model/Course.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
```
Comment (JoseCortes-Agile) [inline src/main/java/com/university/university/model/Course.java]: Please do not use Data annotation it import a lot of classes that we do not use, 
Only use the necessary annotations

File: src/main/java/com/university/university/model/Course.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+@Document(collection = "courses")
+public class Course {
+
+    @Id
+    private String _id;
```
Comment (JoseCortes-Agile) [inline src/main/java/com/university/university/model/Course.java:14]: Id should be String.

File: src/main/java/com/university/university/repository/CourseRepository.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
+package com.university.university.repository;
+
+import java.util.List;
```
Comment (JoseCortes-Agile) [inline src/main/java/com/university/university/repository/CourseRepository.java]: Remove unused imports

File: src/main/java/com/university/university/service/AtUniversityServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Autowired
+    private CourseRepository courseRepository;
+
+    @Override
+    public List<Course> getCourses() {
+        List<Course> courses = new ArrayList<>();
+
+        courses = courseRepository.findAll();
+        return courses;
```
Comment (JoseCortes-Agile) [inline src/main/java/com/university/university/service/AtUniversityServiceImpl.java]: Courses variable is not needed, we can return the list directly in Return statement.

**Review by JoseCortes-Agile — CHANGES_REQUESTED** [review]

**Comment by JoseCortes-Agile** [comment]
Please pull also develop code into your branch, seems that we have new changes there and resolve conflicts

File: src/main/java/com/university/university/controller/UniversityController.java [hunk 8]
```diff
@@ -1,5 +1,7 @@
 package com.university.university.controller;
```
Comment (at-internship) [inline src/main/java/com/university/university/controller/UniversityController.java:1]: please refactor package name to 
package com.agilethought.internship.university.controller;

File: src/main/resources/application.properties [hunk 9]
```diff
@@ -1,2 +1,2 @@
 #server.port=3001
-spring.data.mongodb.uri: mongodb+srv://db_user:vloUr7AakeTEdXDj@cluster0-nomfh.mongodb.net/InternHome?retryWrites=true&w=majority
+spring.data.mongodb.uri: mongodb+srv://at-university-api:4gil3th0ught@cluster0.6xlmy.mongodb.net/at-university-db?retryWrites=true&w=majority
```
Comment (at-internship) [inline src/main/resources/application.properties:2]: This type of sensitive information should not be added here

**Review by at-internship — CHANGES_REQUESTED** [review]
Verify comments

**Comment by JoseCortes-Agile** [comment]
I see the PR has conflicts, please solve them, also it has some comments from At-intership, please review them

**Pull request closed without merging**